nginx-sidecar/handlers/healthcheck_handler.go

```diff
@@ -0,0 +1,43 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+)
+
+func HealthcheckHandler(w http.ResponseWriter, r *http.Request) {
+	urls := r.URL.Query()["url"]
+
+	if len(urls) < 1 {
+		log.Println("missing parameter url")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	for _, urlToCheck := range urls {
+		urlToCheck, err := url.Parse(urlToCheck)
+
+		if err != nil {
+			log.Printf("url format error: %q", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		resp, err := http.Get(urlToCheck.String())
```

Comment:
Calling `http.Get` uses the `http.DefaultClient` instance, which does not have any configured timeout. This could cause this call to block for a long time if there's a problem causing nginx to hang. I think we could use a custom `http.Client` instance with some sensible timeout configured.

Reply:
Sure, I agree. The probe default timeout is 1 second, but even after the probe failure, this call will keep blocked while the Nginx is not responding. I think a timeout of 1 second on this request is enough. What do you think?